`TestPlayerError ` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/28176008420/job/83476793727?pr=67988

#### Relevant snippet

```
=== Failed
=== FAIL: lib/player  (0.00s)
-test.shuffle 1782403384601404362
panic: test timed out after 10m0s
	running tests:
		TestPlayerError (10m0s)

goroutine 130 [running]:
testing.(*M).startAlarm.func1()
	/opt/go/src/testing/testing.go:2802 +0x605
created by time.goFunc
	/opt/go/src/time/sleep.go:215 +0x45

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0xc00079d208, {0x42a3786, 0xf}, 0x44c5d30)
	/opt/go/src/testing/testing.go:2109 +0xb3e
testing.runTests.func1(0xc00079d208)
	/opt/go/src/testing/testing.go:2585 +0x85
testing.tRunner(0xc00079d208, 0xc000857ad0)
	/opt/go/src/testing/testing.go:2036 +0x21d
testing.runTests({0x42d13f7, 0x21}, {0x42f23df, 0x2c}, 0xc0003da3f0, {0x63e2b60, 0xa, 0xa}, {0xc28772c423da446c, 0x8bb9dbcf6b, ...})
	/opt/go/src/testing/testing.go:2583 +0x9ea
testing.(*M).Run(0xc000403cc0)
	/opt/go/src/testing/testing.go:2443 +0xf4c
main.main()
	_testmain.go:66 +0x165

goroutine 126 [chan send, 10 minutes]:
github.com/gravitational/teleport/lib/player_test.TestPlayerError(0xc0008bb688)
	/__w/teleport/teleport/lib/player/player_test.go:100 +0x3e5
testing.tRunner(0xc0008bb688, 0x44c5d30)
	/opt/go/src/testing/testing.go:2036 +0x21d
created by testing.(*T).Run in goroutine 1
	/opt/go/src/testing/testing.go:2101 +0xb13

goroutine 127 [select, 10 minutes]:
github.com/gravitational/teleport/lib/player.(*Player).waitWhilePaused(...)
	/__w/teleport/teleport/lib/player/player.go:451
github.com/gravitational/teleport/lib/player.(*Player).stream(0xc0004734a0, {0x4514c70, 0x6434d40})
	/__w/teleport/teleport/lib/player/player.go:219 +0x44f
created by github.com/gravitational/teleport/lib/player.New in goroutine 126
	/__w/teleport/teleport/lib/player/player.go:168 +0x7c7
FAIL	github.com/gravitational/teleport/lib/player	600.234s

=== FAIL: lib/player TestPlayerError (unknown)
```

on master https://github.com/gravitational/teleport/actions/runs/28268586581/job/83760951951?pr=68023

```
=== FAIL: lib/player  (0.00s)
-test.shuffle 1782513183039257449
panic: test timed out after 10m0s
	running tests:
		TestPlayerError (10m0s)
```

--- a/lib/player/player_test.go
+++ b/lib/player/player_test.go
@@ -87,12 +87,10 @@ func TestPlayPause(t *testing.T) {
 }
 
 func TestPlayerError(t *testing.T) {
-	clk := clockwork.NewFakeClock()
 	errCh := make(chan error)
 	p, err := player.New(&player.Config{
-		Clock:     clk,
 		SessionID: "test-session",
-		Streamer:  &simpleStreamer{count: 1, errCh: errCh},
+		Streamer:  &errorStreamer{errCh: errCh},
 		Log:       slog.New(slog.DiscardHandler),
 	})
 	require.NoError(t, err)
@@ -101,12 +99,10 @@ func TestPlayerError(t *testing.T) {
 
 	_, ok := <-p.C()
 	require.False(t, ok)
-	playerErr := p.Err()
-	// Player.Err() should return the backend error that
-	// caused playback failure.
-	require.Error(t, playerErr)
-	assert.Contains(t, playerErr.Error(), "some error")
 
+	// Player should return the backend error that
+	// caused playback failure.
+	require.ErrorContains(t, p.Err(), "some error")
 }
 
 func TestAppliesTiming(t *testing.T) {
@@ -371,6 +367,16 @@ func (s *simpleStreamer) StreamSessionEvents(ctx context.Context, sessionID sess
 	}()
 
 	return evts, s.errCh
+}
+
+// errorStreamer is a Streamer that never emits any events and only
+// surfaces an error.
+type errorStreamer struct {
+	errCh chan error
+}
+
+func (s *errorStreamer) StreamSessionEvents(ctx context.Context, sessionID session.ID, startIndex int64) (chan apievents.AuditEvent, chan error) {
+	return make(chan apievents.AuditEvent), s.errCh
 }
 
 type databaseStreamer struct {

